media/js/teach/smarton.js

```diff
@@ -0,0 +1,261 @@
+                } else if ((direction === 'up') && (!$body.hasClass('footer-closed'))) {
+                    // slide down when scrolling up, then unstick
+                    $pageFooter.animate({ bottom: '-' + $pageFooter.height() + 'px' }, 500, function() {
+                        $pageFooter.removeClass('stuck').removeAttr('style').find('button.close').remove();
+                    });
+                }
+            }
+        });
+
+        // Unstick the footer when we reach the bottom
+        var unstickyFooter = new Waypoint({
+            element: $('#colophon'),
+            offset: '100%',
+            handler: function(direction) {
+                if ((direction === 'down') && (!$body.hasClass('footer-closed'))) {
+                    $pageFooter.removeClass('stuck').find('button.close').remove();
+                } else if ((direction === 'up') && (!$body.hasClass('footer-closed'))) {
+                    $pageFooter.addClass('stuck').append(buttonClose);
+                    initFooterClose();
+                }
+            }
+        });
+    }
+
+    // Dismiss the sticky footer
+    function initFooterClose() {
+        var footerVisible = $('#footer-cta > .content:visible').data('footer-name');
+        var footerDetails = $('.footer-cta #form-details, .footer-cta .form-details');
+
+        $('.footer-cta .close').on('click', function() {
+            if (tallMode && wideMode) {
+                $pageFooter.animate({ bottom: '-' + $pageFooter.height() + 'px' }, 500, function() {
+                    $pageFooter.removeClass('stuck').removeAttr('style').find('.close').remove();
+                });
+            } else {
+                $pageFooter.removeClass('stuck').removeAttr('style').find('.close').remove();
+            }
+
+            // Close the form if it's open
+            if (footerDetails.is(':visible')) {
+                footerDetails.slideUp('fast');
+            }
+
+            // A class lets us check if the footer was dismissed on purpose
+            // so we don't show it again during scrolling
+            $body.addClass('footer-closed');
+
+            // Track it and record which one was visible
+            window.dataLayer.push({
+                event: 'smarton-interactions',
+                interaction: 'dismiss sticky footer',
+                footer: footerVisible
+            });
+        });
+    }
+
+    // Scroll smoothly to the linked section
+    $document.on('click', '.nav-steps a[href^="#"]', function(e) {
+        e.preventDefault();
+        // Extract the target element's ID from the link's href.
+        var elem = $(this).attr('href').replace(/.*?(#.*)/g, '$1');
+        Mozilla.smoothScroll({
+            top: $(elem).offset().top - $pageHeader.height()
+        }, function() {
+            // Add a tabindex to the target element so it can recieve focus, focus it, then remove tabindex
+            $(elem).attr('tabindex','100').focus().removeAttr('tabindex');
```

Comment:
Re-reading your explanation I think I see where you're coming from, but this will still need to be reworked somehow if you want to keep it, but there's no callback for when a native scroll event has ended. Maybe something we could look to add in the helper.